concrete-core/src/specification/engines/glwe_ciphertext_ggsw_ciphertext_discarding_external_product.rs

```diff
@@ -0,0 +1,81 @@
+use crate::specification::engines::AbstractEngine;
+use crate::specification::entities::{GgswCiphertextEntity, GlweCiphertextEntity};
+
+use super::engine_error;
+
+engine_error! {
+    GlweCiphertextGgswCiphertextDiscardingExternalProductError for
+    GlweCiphertextGgswCiphertextDiscardingExternalProductEngine @
+    PolynomialSizeMismatch => "All the GGSW and GLWE ciphertexts polynomial sizes must be the same.",
+    GlweDimensionMismatch => "All the GGSW and GLWE ciphertexts GLWE dimension must be the same."
+}
```

Comment:
Maybe a check on the secret key?

Reply:
The secret key? It is not needed (or available) during an external product, right?

Reply:
Sorry I was unclear. The secret key is not available, but we can check if the secret key type of the two ciphertexts are the same. I think I asked for this type of check in a previous PR I reviewed and maybe it is not necessary. @aPere3 maybe can help us here.

Reply:
This is actually performed at compile time due to the trait bounds in the definition of the `GlweCiphertextGgswCiphertextDiscardingExternalProductEngine` trait !